Add unit tests for Container AddComponent.

vernonr3 opened the pull request:
Provides unit test of function:
Adding existing component returns existing
Adding new component returns appended component
Adding component that matches name but has update fields tests the merging of those fields. With exception of DSLFunc. Not quite sure why the merging function doesn't work here.
Adds script to assess unit test coverage and html output to get colour coded output.
States unit coverage of file now to be 93.3%..

I haven't sought to minimise the lines of test code & initialisation - aiming to be clear rather than clever.

raphael commented:
This is great, thank you for the nice contribution! A few things before we can merge it:
* The [tests fail](https://github.com/goadesign/model/actions/runs/3976611293/jobs/6818043936) because of linting issues. You should be able to see them locally by running `make` or `make lint`.
* Test coverage should already be reported via Foresight, for example in [this build](https://goa.app.runforesight.com/repositories/github/goadesign/model/workflow-run/f404faf4-952d-3a1e-9abd-fdee066b1684/10623633406/tests). The change impact analysis is empty though, so something isn't quite right - I'll look into it. Still, we probably shouldn't add `assesscoverage` script and `tmp` directory to the repo.

Thanks again, I'll merge once the build passes, and the test coverage stuff is removed. Ideally, we'd also remove the commented-out test (or enable/finish it).

vernonr3 commented:
Thanks Raphael for your warm and quick response.

- I've sorted the linting issues
- Removed the assesscoverage and tmp directories from the repo. I'll use them locally as it gives nice coloured feedback on progress
- I've also completed the commented out test and added another. Do you want a separate PR for this?
Thanks Vernon

raphael commented:
This is great, no need for a separate PR. Thank you!
